### PR TITLE
Reorganize cilium agent CLI flags

### PR DESCRIPTION
**Pull request by aanm**
- Reorganized cilium-agent flags to be in a single place;
- Made function for environment variable names to be consistent for all options used in cilium-agent;
- Decrease dependency of viper library across all code, make use of `pkg/config` instead;
- Print values set for all flags before starting cilium-agent.

Future work (in a following PR):
- Stop setting Cilium options via environment variables in DaemonSet, the ConfigMap already mounts all options in the container. Prototype already in place, needs more testing.

```release-note
Print all CLI flags when starting cilium-agent
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6539)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-K8s/1447/ is being fixed by #6551 
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/8762/ is being fixed by #6552

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-missed-k8s

File: Gopkg.toml [hunk 1]
```diff
@@ -372,21 +372,21 @@ required = [
...
-  revision = "9ff6c6923cfffbcd502984b8e0c80539a94968b7"
+  revision = "24fa6976df40757dce6aea913e7b81ade90530e1"
 
   # main-usage = "mainly CLI packages"
   # on-revision = ""
 
 [[constraint]]
   name = "github.com/spf13/viper"
-  revision = "7538d73b4eb9511d85a9f1dfef202eeb8ac260f4"
+  revision = "v1.3.1"
```
Comment (jrajahalme) [inline Gopkg.toml:389]: Do we have a policy on depending on SHA vs. version tag? The tagged commit could change, so would a dependency to a SHA be better?

File: daemon/daemon.go [hunk 2]
```diff
@@ -547,7 +546,7 @@ func (d *Daemon) init() error {
 		sockops.SockmapDisable()
 		sockops.SkmsgDisable()
 
-		if viper.GetBool(option.SockopsEnableName) == true {
+		if option.Config.SockopsEnable {
```
Comment (jrajahalme) [inline daemon/daemon.go:595]: 👍

File: pkg/option/config.go [hunk 3]
```diff
@@ -216,8 +379,42 @@ func GetTunnelModes() string {
 	return fmt.Sprintf("%s, %s, %s", TunnelVXLAN, TunnelGeneve, TunnelDisabled)
 }
 
-// daemonConfig is the configuration used by Daemon.
-type daemonConfig struct {
+// getEnv returns the environment variable to be used for the given option name.
+func getEnv(option string) string {
```
Comment (jrajahalme) [inline pkg/option/config.go]: In other contexts "getenv" returns the value of the environment variable. Maybe rename this as `getEnvName()` instead?

**Review by jrajahalme — APPROVED** [review]
Small nits only.

File: Gopkg.toml (hunk 1 above)
Comment (aanm) [reply Gopkg.toml:389]: That's a good point indeed :thinking: Can be done in a future PR for all dependencies

**Review by tgraf — APPROVED** [review]
Awesome cleanup. I agree that we should backport into 1.4 as backports will become messy otherwise. I'll merge this as soon as you rebase.

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Pull request merged**